pkg/operation/botanist/controlplane/kube_apiserver_service.go

```diff
@@ -125,16 +149,13 @@ func (d *kubeAPIService) Wait(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
 	defer cancel()
 
-	svcNS := d.service.Namespace
-	svcName := d.service.Name
-
-	if d.values.EnableSNI {
-		svcNS = d.sniService.Namespace
-		svcName = d.sniService.Name
-	}
-
 	return d.waiter.Until(ctx, 5*time.Second, func(ctx context.Context) (done bool, err error) {
-		loadBalancerIngress, err := kutil.GetLoadBalancerIngress(ctx, d.client, svcNS, svcName)
+		loadBalancerIngress, err := kutil.GetLoadBalancerIngress(
+			ctx,
+			d.client,
+			d.loadBalancerServicekey.Namespace,
+			d.loadBalancerServicekey.Name,
+		)
 		if err != nil {
 			d.logger.Info("Waiting until the API Server LoadBalancer deployed in the Seed cluster is ready...")
```

Comment:
It can either be the Shoot API server Load balancer or the Istio Ingress gateway LB.
It would make it clearer if the correct svc is stated.